### PR TITLE
Implement the mechanics of fxa login on android, but don't show ui fo…

### PR DESCRIPTION
**Pull request by fzzzy**
…r it yet. Also, scopedKeys are not yet implemented.

### CODE REVIEW

**Comment by csadilek** [comment]
Hi, Just wanted to give a heads-up that the FxA client API is changing. You can find details in our changelog: https://mozilla-mobile.github.io/android-components/changelog/

Or, probably easier, when you're ready to upgrade, checkout the updated README and the FxA sample app:
https://github.com/mozilla-mobile/android-components/blob/master/components/service/firefox-accounts/README.md

https://github.com/mozilla-mobile/android-components/blob/master/samples/firefox-accounts/src/main/java/org/mozilla/samples/fxa/MainActivity.kt

The main change is that we've switched from supporting a custom promise type (FxaResult) to Kotlin's Deferred, and make heavy use of coroutines. We're still working on the best practices for coroutines in Android components, but the links above should get you to a working solution.

If you need any help, I am happy to help. @thomcc or @vladikoff also have the details.

File: package-lock.json [hunk 1]
```diff
@@ -7016,6 +7016,15 @@
...
     },
+<<<<<<< Updated upstream
+=======
+    "hoek": {
+      "version": "4.2.1",
+      "resolved": "https://registry.npmjs.org/hoek/-/hoek-4.2.1.tgz",
+      "integrity": "sha512-QLg82fGkfnJ/4iy1xZ81/9SIJiq1NGFUMGs6ParyjBZr6jW2Ufj/snDqTHixNlHdPNwN2RLVD0Pi3igeK9+JfA==",
+      "dev": true
+    },
+>>>>>>> Stashed changes
```
Comment (dannycoates) [inline package-lock.json]: :sad_trombone:

**Comment by fzzzy** [comment]
@csadilek Yes, I was aware of that. Did it land? When I looked the PR was still open.

**Comment by csadilek** [comment]
@fzzzy yes, it's part of the 0.30.0 release. I just send this PR to the reference browser to upgrade: https://github.com/mozilla-mobile/reference-browser/pull/172

We'll work on further simplifications in the future, but this should definitely be an improvement (lots of bug fixes and potential leaks got fixed.)

**Pull request merged**